hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/DataSerializableSerializer.java

```diff
@@ -117,7 +118,11 @@ private DataSerializable readInternal(ObjectDataInput in, Class aClass)
         DataSerializable ds = null;
         if (null != aClass) {
             try {
-                ds = (DataSerializable) aClass.newInstance();
+                try {
```

Comment:
I'm not crazy about catching an exception, mutating it and then recatching it.

Reply:
Fixed